pkg/graveler/graveler.go

```diff
@@ -1684,53 +1718,52 @@ func (g *KVGraveler) Commit(ctx context.Context, repositoryID RepositoryID, bran
 		if params.SourceMetaRange != nil {
 			commit.MetaRangeID = *params.SourceMetaRange
 		} else {
-			changes, err := g.StagingManager.List(ctx, branch.StagingToken, ListingMaxBatchSize)
+			changes, err := g.listSealedTokens(ctx, branch)
```

Comment:
My concern here is that if the sealed tokens modified - we can stop trying to commit or do the optimization and remove from the sealed tokens just the tokens we set to commit.
We are using 'retryBranchUpdate' after we updated the branch and we do not checking the specific sealed tokens we set to commit and just list all the latest.

Reply:
It doesn't really matter. Even if some other commit flow succeeded to add another `stagingToken` after you added yours and now you read it, our commit flow can still succeed if no one updated the branch in between.